### PR TITLE
chore(auth): enable lint on __tests__

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

See each commit message for detials.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/__tests__/mockData.ts [hunk 1]
```diff
@@ -1,6 +1,3 @@
-// keeping this so that jest does not complain
-test('should do nothing', () => {});
-
```
Comment (ashika112) [inline packages/auth/__tests__/mockData.ts:3]: 👀

File: packages/auth/__tests__/providers/cognito/credentialsProvider.test.ts [hunk 2]
```diff
@@ -99,9 +98,10 @@ describe('Guest Credentials', () => {
 				{ IdentityId: 'identity-id-test' },
 			);
 			expect(
-				cognitoCredentialsProvider['_nextCredentialsRefresh'],
+				(cognitoCredentialsProvider as any)._nextCredentialsRefresh,
```
Comment (ashika112) [inline packages/auth/__tests__/providers/cognito/credentialsProvider.test.ts:101]: hmm. i think we can remove this line of test. I dont think this helps here. Correct me if im worng. 

Also, i think we shouldn't test private variable, those are supposed to be private to the class implementation. i think as long as we have tested for guest cred i think we are good here.

File: packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts [hunk 3]
```diff
@@ -143,7 +143,7 @@ describe('TokenOrchestrator', () => {
 			mockAuthTokenStore.loadTokens.mockResolvedValue(validAuthTokens);
 			(oAuthStore.loadOAuthInFlight as jest.Mock).mockResolvedValue(true);
 
-			expect(await tokenOrchestrator.getTokens()).resolves;
+			await tokenOrchestrator.getTokens();
```
Comment (ashika112) [inline packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts:146]: could this be `resolves.ToBe()` rather?

**Review by israx — DISMISSED** [review]

**Review by jimblanc — DISMISSED** [review]

File: packages/auth/__tests__/providers/cognito/assertServiceError.test.ts [hunk 4]
```diff
@@ -7,7 +8,9 @@ describe('asserts service errors', () => {
 	test('it should throw an unknown error when error is null', () => {
```
Comment (cshfang) [inline packages/auth/__tests__/providers/cognito/assertServiceError.test.ts:8]: Should these tests with `try`/`catch` clauses have `expect.assertions()`?

File: packages/auth/__tests__/providers/cognito/fetchAuthSession.test.ts [hunk 5]
```diff
@@ -75,9 +75,8 @@ describe('fetchAuthSession behavior for IdentityPools only', () => {
 });
 
 describe('fetchAuthSession behavior for UserPools only', () => {
-	let tokenProviderSpy;
 	beforeEach(() => {
-		tokenProviderSpy = jest
+		jest
```
Comment (cshfang) [inline packages/auth/__tests__/providers/cognito/fetchAuthSession.test.ts:79]: Nit: `beforeAll`?

File: packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts [hunk 6]
```diff
@@ -29,9 +32,9 @@ const authConfig = {
 const mockedGetCurrentUser = getCurrentUser as jest.Mock;
 
 describe('confirmSignIn API happy path cases', () => {
-	let handleChallengeNameSpy;
-	const username = authAPITestParams.user1.username;
-	const password = authAPITestParams.user1.password;
+	let handleChallengeNameSpy: jest.SpyInstance;
+	const { username } = authAPITestParams.user1;
+	const { password } = authAPITestParams.user1;
```
Comment (cshfang) [inline packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts:37]: Nit: Here and below and in other files
```suggestion
	const { password, username } = authAPITestParams.user1;
```

**Review by cshfang — APPROVED** [review]

File: packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts (hunk 6 above)
Comment (ashika112) [reply packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts:37]: this might have been auto fix run.

Comment (HuiSF) [reply packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts:37]: Hum I'm if there is a lint rule for this and auto fixable, will look

File: packages/auth/__tests__/providers/cognito/assertServiceError.test.ts (hunk 4 above)
Comment (HuiSF) [reply packages/auth/__tests__/providers/cognito/assertServiceError.test.ts:8]: I rewrote these tests to use `.toThrow()` to assert the expected error to be thrown.

File: packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts (hunk 6 above)
Comment (HuiSF) [reply packages/auth/__tests__/providers/cognito/confirmSignInHappyCases.test.ts]: Didn't find a rule for this for auto fix. Manually fixed.

File: packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts (hunk 3 above)
Comment (HuiSF) [reply packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts]: This `await tokenOrchestrator.getTokens();` is actually not needed. Refactored

File: packages/auth/__tests__/providers/cognito/credentialsProvider.test.ts (hunk 2 above)
Comment (HuiSF) [reply packages/auth/__tests__/providers/cognito/credentialsProvider.test.ts:101]: Agreed, but I don't want to refactor the original tests "too much".

File: packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts [hunk 7]
```diff
@@ -143,15 +143,9 @@ describe('TokenOrchestrator', () => {
...
 			(oAuthStore.loadOAuthInFlight as jest.Mock).mockResolvedValue(true);
 
-			await tokenOrchestrator.getTokens();
+			const tokens = await tokenOrchestrator.getTokens();
 
 			expect(addInflightPromise).toHaveBeenCalledWith(expect.any(Function));
-			expect(mockAddInflightPromise).toHaveBeenCalledTimes(1);
-
-			const inflightPromiseResolver = mockAddInflightPromise.mock.calls[0][0];
-			inflightPromiseResolver();
```
Comment (HuiSF) [inline packages/auth/__tests__/providers/cognito/tokenOrchestrator.test.ts:155]: This is actually not needed as the `mockAddInflightPromise` is already resolving the added promise in the mock.

**Review by ashika112 — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Review by israx — APPROVED** [review]

**Pull request merged**